data_structures/src/chain.rs

```diff
@@ -2559,15 +2586,48 @@ impl AltKeys {
     pub fn insert_keys_from_block(&mut self, block: &Block) {
         // Add miner bn256 public keys
         if let Some(value) = block.block_header.bn256_public_key.clone() {
-            self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+            // If the provided key is valid, insert it in altkeys
+            if value.is_valid() {
```

Comment:
should this validation be done inside of the insert_bn256 method?

Reply:
makes sense